Script for process real world abilene traffic traces

RafaelSche opened the pull request:
Belongs to https://github.com/RealVNF/coord-sim/issues/113

Content:
- convert_traces.py:
    - contains class TraceXMLReader which holds all the parameters and data
    - its three main functions are:
        - read_one_file(), which reads one xml file, finds the relevant data and stores it in a object attribute dataframe
        - read_files_parallel(), which finds a certain number or all the files in a given directory, calling os.listdir(), and starts threads for every file. Also it stores the intermediate results to a csv file, in order to reuse them, since the mentioned step take the longest time
        - process_df(), which processes the df with the intermediate results. Means it apply the scale_factor and converts the data rate into inter_arrival_mean. Also the choice of ingress nodes is made here and the time axis, which is defined by run_duration and change_rate is calculated

- trace_xml_reader_config.yaml: Example of a configuration
- node_name_map: defines what names of nodes should be in the trace (if you set a node to null (None in Python) it will be removed)

stefanbschneider commented:
Looks good, good work! Did you test it and is it working? What's still missing?

To me it looks good. Regarding documentation, please

* Add docstrings in your script to briefly explain what each function does and what its parameters mean.
* Add a short subsection to the readme explaining how to use and convert real-world traces. What are the exact steps? Where to download the XML files, how to configure & use the converter, how to use the resulting, converted traces. Add it as subsection under "Usage" https://github.com/RealVNF/coord-sim#usage

RafaelSche commented:
@stefanbschneider, yes, that is what I wanted to do. I also added plotting and found some small bugs. I added the section in the Readme. Now it should be ready after review.
